Bug:  Create own struct to be able to add omitempty json tag on attributes

stromp opened the pull request:
### Description
The whole struct was returned which marshalled attributes with a null values. Those attributes should be omitted, since you can not use them when you provider the json to AWS. 

### Relations
Relates #33565

### Output from Acceptance Testing
=== RUN   TestAccSSMPatchBaseline_basic
=== PAUSE TestAccSSMPatchBaseline_basic
=== CONT  TestAccSSMPatchBaseline_basic
--- PASS: TestAccSSMPatchBaseline_basic (34.30s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/ssm        34.398s
...
```

justinretzolk commented:
Hey @stromp 👋 Thank you for your contribution here! I'm not one of the people who will be doing a final review on this, but I wanted to note that I added the `enhancement` label rather than the `bug` one, since this is a change to make the resource more friendly to use, and not something that causes the provider to throw errors. If I've misunderstood, please do let me know and we can update the labels.

stromp commented:
> Hey @stromp 👋 Thank you for your contribution here! I'm not one of the people who will be doing a final review on this, but I wanted to note that I added the `enhancement` label rather than the `bug` one, since this is a change to make the resource more friendly to use, and not something that causes the provider to throw errors. If I've misunderstood, please do let me know and we can update the labels.

hi @justinretzolk.

I guess it is a bit how you look at it. Currently the resource will give you a different JSON result then the original JSON returned by the AWS API would. Since the purpose of the attribute is that it returns the API JSON I would personally classify this as a bug.

stromp commented:
> > Hey @stromp 👋 Thank you for your contribution here! I'm not one of the people who will be doing a final review on this, but I wanted to note that I added the `enhancement` label rather than the `bug` one, since this is a change to make the resource more friendly to use, and not something that causes the provider to throw errors. If I've misunderstood, please do let me know and we can update the labels.
> 
> hi @justinretzolk.
> 
> I guess it is a bit how you look at it. Currently the resource will give you a different JSON result then the original JSON returned by the AWS API would. Since the purpose of the attribute is that it returns the API JSON I would personally classify this as a bug.

hi @justinretzolk

I have just rebased the branch since it became out of sync due to the aws sdk update. Can I ask for some attention to get this merged?

jar-b reviewed (approved):
LGTM 🎉 

```console
% make testacc PKG=ssm TESTS=TestAccSSMPatchBaseline_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.2 test ./internal/service/ssm/... -v -count 1 -parallel 20 -run='TestAccSSMPatchBaseline_'  -timeout 360m

--- PASS: TestAccSSMPatchBaseline_disappears (31.31s)
=== CONT  TestAccSSMPatchBaseline_tags_null
--- PASS: TestAccSSMPatchBaseline_rejectPatchesAction (37.77s)
=== CONT  TestAccSSMPatchBaseline_tags_DefaultTags_providerOnly
--- PASS: TestAccSSMPatchBaseline_approvedPatchesNonSec (39.94s)
=== CONT  TestAccSSMPatchBaseline_tags_DefaultTags_updateToProviderOnly
--- PASS: TestAccSSMPatchBaseline_tags_DefaultTags_nullOverlappingResourceTag (43.85s)
=== CONT  TestAccSSMPatchBaseline_tags_DefaultTags_updateToResourceOnly
--- PASS: TestAccSSMPatchBaseline_tags_DefaultTags_nullNonOverlappingResourceTag (43.91s)
=== CONT  TestAccSSMPatchBaseline_tags_DefaultTags_overlapping
--- PASS: TestAccSSMPatchBaseline_tags_DefaultTags_emptyProviderOnlyTag (45.51s)
=== CONT  TestAccSSMPatchBaseline_tags_EmptyTag_OnUpdate_Replace
--- PASS: TestAccSSMPatchBaseline_tags_ComputedTag_OnCreate (49.76s)
--- PASS: TestAccSSMPatchBaseline_tags_DefaultTags_emptyResourceTag (50.56s)
--- PASS: TestAccSSMPatchBaseline_operatingSystem (56.32s)
--- PASS: TestAccSSMPatchBaseline_sources (59.38s)
--- PASS: TestAccSSMPatchBaseline_approveUntilDateParam (59.69s)
--- PASS: TestAccSSMPatchBaseline_approvalRuleEmpty (67.48s)
--- PASS: TestAccSSMPatchBaseline_basic (67.74s)
--- PASS: TestAccSSMPatchBaseline_tags_AddOnUpdate (68.98s)
--- PASS: TestAccSSMPatchBaseline_tags_ComputedTag_OnUpdate_Replace (70.36s)
--- PASS: TestAccSSMPatchBaseline_tags_EmptyTag_OnCreate (72.92s)
--- PASS: TestAccSSMPatchBaseline_tags_null (42.16s)
--- PASS: TestAccSSMPatchBaseline_tags_ComputedTag_OnUpdate_Add (75.00s)
--- PASS: TestAccSSMPatchBaseline_tags_EmptyTag_OnUpdate_Add (85.97s)
--- PASS: TestAccSSMPatchBaseline_tags_DefaultTags_updateToResourceOnly (42.60s)
--- PASS: TestAccSSMPatchBaseline_tags_DefaultTags_updateToProviderOnly (46.77s)
--- PASS: TestAccSSMPatchBaseline_tags_EmptyTag_OnUpdate_Replace (41.84s)
--- PASS: TestAccSSMPatchBaseline_tags_DefaultTags_nonOverlapping (89.47s)
--- PASS: TestAccSSMPatchBaseline_tags (100.06s)
--- PASS: TestAccSSMPatchBaseline_tags_DefaultTags_overlapping (59.82s)
--- PASS: TestAccSSMPatchBaseline_tags_DefaultTags_providerOnly (77.53s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/ssm        121.254s
```

jar-b commented:
Thanks for your contribution, @stromp! 👍
